src/csv_parser.c

```diff
@@ -0,0 +1,309 @@
+		for(int j=0;j<COL;j++){
+			printf("%f |", csv[i][j]);
+		}
+		printf("\n");
+	}
+    /* Identify unique time stamps and compute number of sensor readings in each unique time stamp
+    * and store all the sensor readings into a 2D array where each row represents one time stamp
+    * For ex: sr[0][] row will contain all sensor readings for 1st time stamp of the csv file
+    * sr[1][] will contain all sensor readings for 2nd time stamp
+    * sr_c[] will contain number of sensors in each time stamp. For ex: value stored in ts_c[0] 
+    * represents number of sensor readings in time stamp0
+	* Extract sensor readings for time stamp 0 and store in sr[0][]
+	* u_ts points to position of unique time stamps in csv[][]
+	*/
+	for(int i=0;i<ROWS;i++){
+		if (csv[i][0]== csv[0][0]){
+			sr[0][i] = csv[i][2];
+			sr_c[0]++;
+			u_ts[0] = sr_c[0];
+		}
+		/* Extract sensor eadings for time stamp 1 and store in sr[1][]*/
+		else if (csv[i][0]== csv[u_ts[0]][0]){
+			sr[1][sr_c[1]] = csv[i][2];
+			sr_c[1]++;
+			u_ts[1] = sr_c[0]+sr_c[1];
+		}
+		/* Extract sensor eadings for time stamp 2 and store in sr[2][]*/
+		else if (csv[i][0]== csv[u_ts[1]][0]){
+			sr[2][sr_c[2]] = csv[i][2];
+			sr_c[2]++;
+			u_ts[2] = sr_c[0]+sr_c[1]+sr_c[2];
+		}
+		/* Extract sensor eadings for time stamp 3 and store in sr[3][]*/
+		else if (csv[i][0]== csv[u_ts[2]][0]){
+			sr[3][sr_c[3]] = csv[i][2];
+			sr_c[3]++;
+			u_ts[3] = sr_c[0]+sr_c[1]+sr_c[2]+sr_c[3];
+		}
+		/* Extract sensor eadings for time stamp 4 and store in sr[4][]*/
+		else if (csv[i][0]== csv[u_ts[3]][0]){
+			sr[4][sr_c[4]] = csv[i][2];
+			sr_c[4]++;
+			u_ts[4] = sr_c[0]+sr_c[1]+sr_c[2]+sr_c[3]+
+			       sr_c[4];
+		}
+		/* Extract sensor eadings for time stamp 5 and store in sr[5][]*/
+		else if (csv[i][0]== csv[u_ts[4]][0]){
+			sr[5][sr_c[5]] = csv[i][2];
+			sr_c[5]++;
+			u_ts[5] = sr_c[0]+sr_c[1]+sr_c[2]+sr_c[3]+
+			       sr_c[4]+sr_c[5];
+		}
+		/* Extract sensor eadings for time stamp 6 and store in sr[6][]*/
+		else if (csv[i][0]== csv[u_ts[5]][0]) {
+			sr[6][sr_c[6]] = csv[i][2];
+			sr_c[6]++;
+			u_ts[6] = sr_c[0]+sr_c[1]+sr_c[2]+sr_c[3]+
+				   sr_c[4]+sr_c[5]+sr_c[6];
+		}
+		/* Extract sensor eadings for time stamp 7 and store in sr[7][]*/
+		else if (csv[i][0]== csv[u_ts[6]][0]) {
+			sr[7][sr_c[7]] = csv[i][2];
+			sr_c[7]++;
+			u_ts[7] = sr_c[0]+sr_c[1]+sr_c[2]+sr_c[3]+
+				   sr_c[4]+sr_c[5]+sr_c[6]+sr_c[7];
+		}
+		/* Extract sensor eadings for time stamp 8 and store in sr[8][]*/
+		else if (csv[i][0]== csv[u_ts[7]][0]) {
+			sr[8][sr_c[8]] = csv[i][2];
+			sr_c[8]++;
+			u_ts[8] = sr_c[0]+sr_c[1]+sr_c[2]+sr_c[3]+
+				   sr_c[4]+sr_c[5]+sr_c[6]+sr_c[7]+
+				   sr_c[8];
+		}
+		/* Extract sensor eadings for time stamp 9 and store in sr[9][]*/	
+		else if (csv[i][0]== csv[u_ts[8]][0]) {
+			sr[9][sr_c[9]] = csv[i][2];
+			sr_c[9]++;
+			u_ts[9] = sr_c[0]+sr_c[1]+sr_c[2]+sr_c[3]+
+			       sr_c[4]+sr_c[5]+sr_c[6]+sr_c[7]+
+			       sr_c[8]+sr_c[9];
+			}
+	}
```

Comment:
This is a lot of generally repeat code that shouldn't really be in a main function. This could easily be transferred to its own function.

Reply:
This was the fundamental function of csv parsing, will be coded as a function where it can be called from main(), However, fixed this in the latest dev branch pull request.